assets/tests/components/ladderPage.test.tsx

```diff
@@ -525,6 +532,52 @@ describe("LadderPage", () => {
     )
     expect(result.queryByText("Vehicles")).toBeNull()
   })
+
+  test("opens the detour modal", async () => {
+    jest
+      .mocked(getTestGroups)
+      .mockReturnValue([
+        TestGroups.DetourRouteSelection,
+        TestGroups.DetoursPilot,
+        TestGroups.RouteLadderHeaderUpdate,
+      ])
+
+    const mockState = {
+      ...initialState,
+      routeTabs: [
+        routeTabFactory.build({
+          selectedRouteIds: ["1"],
+          ordering: 0,
+          isCurrentTab: true,
+        }),
+      ],
+    }
```

Comment:
suggeston: it's inconsitent in this file, but we should be using the factory `stateFactory` to generate the global state.

Additionally, but not required, you shouldn't need to specify ordering because the `routeTabFactory` handles that AFAIK.

```suggestion
    const mockState = stateFactory.build({
      routeTabs: [
        routeTabFactory.build({
          selectedRouteIds: ["1"],
          isCurrentTab: true,
        }),
      ],
    })
```

(non-blocking, just a nit, don't feel like you have to) I'd personally replace the `"1"` route id with `routeFactory.build().id` but I figure you'd prefer the explicitness 😜

Reply:
LOL this comment and https://github.com/mbta/skate/pull/2652/files#r1631385371 are beautiful mirror images of each other